src/blockdata/script.rs

```diff
@@ -122,33 +122,36 @@ pub enum Error {
     /// Tried to read an array off the stack as a number when it was more than 4 bytes
     NumericOverflow,
     /// Error validating the script with bitcoinconsensus library
-    #[cfg(feature = "bitcoinconsensus")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "bitcoinconsensus")))]
-    BitcoinConsensus(bitcoinconsensus::Error),
+    BitcoinConsensus(BitcoinConsensusError),
     /// Can not find the spent output
-    #[cfg(feature = "bitcoinconsensus")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "bitcoinconsensus")))]
     UnknownSpentOutput(OutPoint),
     /// Can not serialize the spending transaction
-    #[cfg(feature = "bitcoinconsensus")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "bitcoinconsensus")))]
     SerializationError
 }
 
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Debug, Clone, Copy)]
+enum Uninhabited {}
+
+#[cfg(feature = "bitcoinconsensus")]
+type BitcoinConsensusError = bitcoinconsensus::Error;
+
+/// Dummy error type used when `bitcoinconsensus` feature is not enabled.
+#[cfg(not(feature = "bitcoinconsensus"))]
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Debug, Clone, Copy)]
```

Comment:
For the backcompat trick to work you may only implement [traits that `bitcoinconsensus::Error` implements](https://docs.rs/bitcoinconsensus/latest/bitcoinconsensus/enum.Error.html#implementations)

Reply:
Regarding derives on dumb error types: while we should not underderive, we can overderive. I.e., if the upstream adds derives to the respective error types this will not be a breaking change for them, but it will break our code. Thus, I suggest to do all possible derives for dumb errors, namely Copy, Eq, PartialEq, Ord, PartialOrd, Hash, Debug (Default for errors is a bad practice, I assume).

Reply:
@dr-orlovsky I think you're wrong here because the feature gate removes our type and replaces it with `bitcoinconsensus::Error`, meaning that if we overderive, the feature-gate will *remove* derives and potentially cause compilation breakage.

Reply:
This is tested in CI in this project, while underderivation may break downstream without CI in this project detecting it (when upstream adds more derives to the error type).

Reply:
Ha! This discussion is mute now, the link above is to `bitcoinconsensus` v0.17.1 but we are depending on v0.19.0-3 which has all the derives on it that we want.

Reply:
My bad for blindly following web rendered docs without checking the version or the code :(

Reply:
Good to have you back man, hope you get fully recovered soon.